classes/Village.php

```diff
@@ -129,4 +135,287 @@ public function getRelations(System $system, string $village_id): array {
         }
         return $relations;
     }
+
+    /**
+     * @return array
+     */
+    public static function getVillagePopulation(System $system, string $village): array {
+        $villager_counts = [];
+        $villager_result = $system->db->query(
+            "SELECT
+                COUNT(IF(`rank`=1,1,NULL)) as `academy`,
+                COUNT(IF(`rank`=2,1,NULL)) as `genin`,
+                COUNT(IF(`rank`=3,1,NULL)) as `chuunin`,
+                COUNT(IF(`rank`=4,1,NULL)) as `jonin`
+                FROM `users` WHERE `village` = '{$village}'"
+        );
+		$villager_result = $system->db->fetch($villager_result);
+        foreach ($villager_result as $key => $value) {
+            $villager_counts[] = [
+                "rank" => $key,
+                "count" => (int)$value,
+            ];
+        }
+        return $villager_counts;
+    }
+
+    /**
+     * @return array
+     */
+    public static function getVillagePopulationTotal(System $system, string $village): int
+    {
+        $villager_result = $system->db->query("SELECT COUNT(*) AS count FROM `users` WHERE `village` = '{$village}'");
+        $villager_result = $system->db->fetch($villager_result);
+        return $villager_result['count'];
+    }
+
+    /**
+     * @return bool
+     */
+    public static function claimSeat(System $system, User $player, string $seat_type): string {
+        switch ($seat_type) {
+            case 'kage':
+                // check seat available
+                $result = $system->db->query("SELECT * FROM `village_seats` WHERE `village_id` = {$player->village->village_id} AND `seat_type` = '{$seat_type}' AND `seat_end` IS NULL LIMIT 1");
```

Comment:
I wonder if it's worth saving the current kage in a single field in the `Village` class so there's always one authoritative "this is the kage" value.

Hypothetically as long as village operations are wrapped in transactions we shouldn't end up in a state where there are two `village_seats` rows with no end, but having it in one column completely prevents that. We could still use this table to track past kages and how long the current kage has been in seat

Reply:
So adding a `kage_id` column to `villages` basically?

Reply:
Yup!